src/App.js

```diff
@@ -30,39 +30,115 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState(null);
+  const [squaresFilled, setSquaresFilled] = useState(0);
 
   // Wave 2
-  // You will need to create a method to change the square 
+  // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
+  const changeSquares = (id) => {
+    if (winner === null) {
+    const newSquares = [...squares];
+
+    // could I repeat the same logic I used to make a 1D array in Board here?
+    // seems redundant that we repeat the logic...could I make a helper function?
+    newSquares.forEach(board => {
+      board.forEach(square => {
+        if (square.id === id && square.value === '') {
+          square.value = player
+          if (player === PLAYER_1) {
+            setPlayer(PLAYER_2)
+            setSquaresFilled(squaresFilled + 1)
+          } else if (player === PLAYER_2) {
+            setPlayer(PLAYER_1)
+            setSquaresFilled(squaresFilled + 1)
+          }
```

Comment:
You can condense this a bit.

```suggestion
        if (square.id === id && square.value === '') {
          square.value = player;
          setPlayer(player === PLAYER_! ? PLAYER_2 : PLAYER_1);
          setSquaresFilled(squaresFilled + 1);
        }
```